Remove overheads and update dependencies

GitBruno opened the pull request:
- Remove `tidy-markdown` (tidy markdown ourselves)
- Remove `xo` (Too much overhead for a small project)
- tidy package.json

We are alive again!

Hypercubed commented:
Are we dead?

Thanks for the PR... this is a lot of changes to one PR and not sure I want to accept all of them.  I'll try to merge soon peice by peice.

GitBruno commented:
Thanks! I can split them up if you like. (I checked out the dev-branch, I can re-do this with Master and split specifics).

This package was not working on the latest version of Node for me. (From memory, tidy-markdown failed.)

So there are only two things, rest can be ignored:
  1. Removed `tidy-markdown` as a dependency (This might be good or bad, but needs work either way)
  2. Added fencing option

GitBruno commented on .gitignore:
```diff
@@ -1 +1,3 @@
+.DS_Store
 node_modules
+package-lock.json
```
Maybe you want this file? but it was not in the original package.

GitBruno commented on .travis.yml:
```diff
@@ -2,4 +2,3 @@ language: node_js
 node_js:
   - "4"
   - "6"
-  - "0.10"
```
Updated dependencies don't run in this version.

GitBruno commented on CHANGELOG.md:
```diff
@@ -1,17 +0,0 @@
---------------------
-* Moved markdown formmater from tap-summary to tap-markdown
-
-undefined
```
I removed this file as it seems not be in use?

GitBruno commented on LICENSE:
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016 Jayson Harshbarger
```
From dev-branch and readme, it is probably a good idea to choose one or the other.

GitBruno commented on README.md:
```diff
@@ -8,7 +8,10 @@ Consume TAP, produce markdown.
 
 ## Install
 
-npm install tap-markdown -g
```
We should not need to install globally

GitBruno commented on README.md:
```diff
@@ -8,7 +8,10 @@ Consume TAP, produce markdown.
-npm install tap-markdown -g
+    npm install tap-markdown
+
+> If you must support Node < 4 you can explicitly install older version of tap-markdown by running the following `npm install tap-markdown@1.2.1`
```
This is a node reflects where we are in light of the updated dependencies.

GitBruno commented on README.md:
```diff
@@ -21,7 +24,7 @@ or in `package.json`
 {
   "scripts": {
-    "test": "tape test/*.js | tap-markdown"
+    "test": "tape test/*.js | tap-markdown > results.md"
```
Add a sample that shows you how to write out to a file.

GitBruno commented on README.md:
````diff
@@ -43,15 +46,17 @@ or in `package.json`
+```	
 
-```
-  --no-tidy        Disable tidy-markdown
````
Maybe we should have this option still?

GitBruno commented on README.md:
````diff
@@ -43,15 +46,17 @@ or in `package.json`
+	
+```	
+  --duration       Disable duration
+  --fence          Enable fencing      
````
Enable to make it backward compatible?

GitBruno commented on test/fixtures/comment.expected:
````diff
@@ -23,3 +23,4 @@ Hello,
 ```
 world!
 ```
+
````
This is annoying, not sure if there is an elegant way of fixing this?

GitBruno commented on index.js:
```diff
@@ -57,32 +63,27 @@ Formatter.prototype.summary = function (summary) {
+  output.push('# Comments' + LF);
   output.push(Object.keys(comments).map(function (name) {
-    return '## ' + name + LF + comments[name].join(LF);
+    return '## ' + name + LF + LF + fence + LF + comments[name].join(LF).replace('    ','') + LF + fence;
```
Should replace contain a smarter regex?

GitBruno commented:
Hey Jayson, how are you going with this? I want to publish a module with this package as a dependency. If you don't have time for this that is OK, I'll find another way to include a copy.

GitBruno commented:
@Hypercubed Is there anything I can do to help?

Hypercubed commented:
Sorry, I haven't had time to look.  It's hard to review code for a project I haven't actively used in a while.
